tests/test_data_samples_order.py

```diff
@@ -0,0 +1,97 @@
+import random
+import pytest
+import copy
+
+
+ALGO_SCRIPT = """
+import json
+import substratools as tools
+class TestAlgo(tools.Algo):
+    def train(self, X, y, models, rank):
+        data_sample_keys = [folder.split('/')[-1] for folder in X]
```

Comment:
I'm not 100% sure you should base the test on the folder name that is currently equal to the data samples hash. This could change later.

IMO the best from a testing perspective would be to load the data values and check the order is as expected for instance. But not sure this is easy to do with the current substra-tests implementation.

Reply:
I went for the most straightforward solution for now, but I agree that in the future we should fond a more durable solution. But as long as we don't have a mechanism that enforces the relative order I'm not sure it's worth the hassle.

Reply:
Sure!